src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -73,7 +76,7 @@
                     String.valueOf(STARTUP_TIME_DEFAULT_VALUE)), STARTUP_TIME_DEFAULT_VALUE);
 
     @DataBoundConstructor
-    public EC2RetentionStrategy(String idleTerminationMinutes) {
+    public EC2RetentionStrategy(String idleTerminationMinutes, String minUpHours) {
```

Comment:
You need to keep the old constructors and mark them `@Deprecated` in order to preserve backward compatibility as this class is public.

Also, you should use an argument type of either `int` (for a required value) or `Integer` (for a nullable one). I'd suggest `OptionalInt` as an argument type for the latter, though I'm not sure if that actually works with `@DataBoundConstructor`.